collectors/ebpf.plugin/ebpf.c

```diff
@@ -633,6 +638,557 @@ static inline int parse_disable_apps(char *ptr)
     return 0;
 }
 
+/**
+ * Fill Port list
+ *
+ * @param out a pointer to the link list.
+ * @param in the structure that will be linked.
+ */
+static inline void fill_port_list(ebpf_network_viewer_port_list_t **out, ebpf_network_viewer_port_list_t *in)
+{
+    if (likely(*out)) {
+        ebpf_network_viewer_port_list_t *move = *out;
+        for (; move->next ; move = move->next ) {
+            if (move->first <= in->first && move->last >= in->first &&
+                move->first <= in->last && move->last >= in->last) {
+                info("The range/value (%u, %u) is inside the range/value (%u, %u) already inserted, it will be ignored.",
+                     ntohs(in->first), ntohs(in->last), ntohs(move->first), ntohs(move->last));
+                freez(in->value);
+                freez(in);
+                return;
+            }
+        }
+
+        move->next = in;
+    } else {
+        *out = in;
+    }
+
+#ifdef NETDATA_INTERNAL_CHECKS
+    info("Adding values %s( %u, %u) to port list used on network viewer",in->value, ntohs(in->first), ntohs(in->last));
+#endif
+}
+
+/**
+ * Fill port list
+ *
+ * Fill an allocated port list with the range given
+ *
+ * @param out a pointer to store the link list
+ * @param range the informed range for the user.
+ */
+static void parse_port_list(void **out, char *range)
```

Comment:
Behaviour here is inconsitent and I think this must be fixed:

1. If I put into list `-111` it will parse it as range `0-111` or by another words `-X`==`NETDATA_MINIMUM_PORT_VALUE-X`
2. But if I put into list `111-` it will parse it as single port `111`, to be consistent with point 1 it should parse as `X-NETDATA_MAXIMUM_PORT_VALUE`

So either such range should in both case give error or we should define it means until the port limit (min/max). (another option would be force it to be single port but I am strongly against this option).

Additionally if I put range `111-100` it will parse first to be `111` and last `100` there seems to be no check if `last>first`. In such case again either we should put error, or put warning and swap the values?

Reply:
additionally if I configure `included ports = 19999 - 12` the second parameter `-` will be parsed as range `0-0` this should be error instead.

Reply:
> In such case again either we should put error, or put warning and swap the values?

i think warning is better, lets dont think for a user (`111-100` - it is likely he wanted `111-1000` and not `100-111`), if there is a mistake, log an error and that is it.

Reply:
@underhood when the first port is not specified I decided to ignore it, because we do not know exactly what the first value the user wants. We could assume the first possible value (`1`), but I am not sure if this is safe.
About the inverted order,  I am not discard the value, instead I am inverting it and writing a message inside the log.